herbert_core/utilities/classes/@hashable/private/to_hashable_array_.m

```diff
@@ -0,0 +1,65 @@
+function [obj,arr] = to_hashable_array_ (obj)
+% Retrieve information specifying
+%
+%   >>[obj,S] = to_hashable_array_(obj)
+%
+% Input:
+% ------
+%   obj -- Object or array of objects which are hashable
+%
+%
+% Output:
+% -------
+%  obj  --  input object, containing possible children hashable objects
+%           wich calculated hashes
+%  arr  --  uint8 array of unique data -- basis for building unique object
+%           hash.
+
+
+% Get hasing fields, used for extracting values, explicitly specifying
+% class state
+field_names = hashableFields (obj(1));
+
+% Recursively turn serializable fields into structures
+arr = cell (1,numel(field_names)*numel(obj));
+ic = 0;
+for j = 1:numel(obj)
+    obj_tmp = obj(j);   % get pointer to jth object to save expensive indexing
```

Comment:
Matlab objects are referenced by `cow` pointers so it is probably a pointer.

Reply:
its not a reference -- it does not fully adhere to reference behaviour

Reply:
Just sometimes it would be good if you made corrections to and expansions of comments in the code when it is clear they don't amke sense to the reader, rather than argue it out in the git comments

Reply:
well, it is a cow pointer. Reference, handle or variable copy all would behave differently (probably closest to variable copy as it will happen eventually later according to the code logic). This is why I am arguing. Would add cow to the explanation.